lib/W3Buffer.ts

```diff
@@ -0,0 +1,64 @@
+import roundTo from 'round-to';
+
+export class W3Buffer {
+
+    public _offset = 0;
+    public _buffer: Buffer;
+
+    constructor(buffer: Buffer) {
+        this._buffer = buffer;
+    }
+
+    readInt() {
```

Comment:
Return types on these `read` functions would be nice: `readInt(): number { ... }`

Reply:
Added in -#3 also added a note for the binary type since this can be anything. Maybe use Buffer.js?